custom_navbutton_completion.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Created by ishani on 01/03/2017.
+ */
+
+custom_navbutton_completion = {
+
+    init: function (Y) {
+
+        var handle_success = function (id, o, args) {
+
+            if (o.responseText != 'OK') {
+                alert('An error occurred when attempting to mark your activity complete.\n\n(' + o.responseText + '.)'); //TODO: localize
+
+            } else {
+                var current = args.state.get('value');
+                var btntype = args.btntype.get('value');
+                var modulename = args.modulename.get('value'),
+                    altstr,
+                    titlestr;
+                if (current == 1) {
+                    // Successfully marked as complete, so change button to 'Mark incomplete'
+                    altstr = M.util.get_string('incompletebuttontext', 'block_navbuttons');
+                    titlestr = M.util.get_string('incompletebuttontext', 'block_navbuttons');
+                    args.state.set('value', 0);
+                    if(btntype == 'icon') {
+                        args.image.set('src', M.util.image_url('crossicon', 'block_navbuttons'));
+                        args.image.set('alt', altstr);
+                        args.image.set('title', titlestr);
+                    }
+                    args.submit.set('value', altstr);
+                } else {
+                    altstr = M.util.get_string('completebuttontext', 'block_navbuttons');
+                    titlestr = M.util.get_string('completebuttontext', 'block_navbuttons');
+                    args.state.set('value', 1);
+                    if(btntype == 'icon') {
+                        args.image.set('src', M.util.image_url('tickicon', 'block_navbuttons'));
+                        args.image.set('alt', altstr);
+                        args.image.set('title', titlestr);
+                    }
+                    args.submit.set('value', altstr);
+                }
+            }
+
+            args.ajax.remove();
+        };
+
+        var handle_failure = function (id, o, args) {
+            alert('An error occurred when attempting to mark your activity complete.\n\n(' + o.responseText + '.)'); //TODO: localize
```

Comment:
Would be better as a Moodle notification popup, rather than an alert message (but, as it should never really appear, I'm not going to worry too much).